src/openrct2/world/Banner.cpp

```diff
@@ -128,6 +128,27 @@ TileElement* banner_get_tile_element(BannerIndex bannerIndex)
     return nullptr;
 }
 
+TileElement* banner_get_scrolling_wall_tile_element(BannerIndex bannerIndex)
+{
+    rct_banner* banner = &gBanners[bannerIndex];
+    TileElement* tileElement = map_get_first_element_at(banner->x, banner->y);
+
+    do
+    {
+        if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
+            continue;
```

Comment:
You can improve the code by only casting to wall once, the As methods are type safe.
```
auto wallElement = tileElement->AsWall();
if (wallElement != nullptr)
{
    auto sceneryEntry = wallElement->GetEntry();
    // etc.
}
```